### ISSUE TITLE
Work around SQLite INSERT .. SELECT .. ON CONFLICT bug

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13468]
SQLite 3.36.0.3 has the following bug:

Given this schema:

```sql
create table t (i int primary key, j int);
```

The following statement doesn't work:

```sql
insert into t (i, j)
select 1, 2
from (values (1))
on conflict do nothing;
```

The error is:

```
SQL Error [1]: [SQLITE_ERROR] SQL error or missing database (near "do": syntax error)
```
Removing the superfluous (in the example) FROM clause helps work around the problem. This works:

```sql
insert into t (i, j)
select 1, 2
on conflict do nothing;
```

Adding a dummy WHERE clause also helps work around the problem:

```sql
insert into t (i, j)
select 1, 2
from (values (1))
where true
on conflict do nothing;
```

So, the most generic workaround from the jOOQ side would be to generate this dummy `WHERE TRUE` clause whenever there's a `FROM` clause in `INSERT .. SELECT .. ON CONFLICT`

**Comment by lukaseder** [comment]
I've reported the bug also to SQLite:
https://www.sqlite.org/forum/forumpost/d03b7325bd

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.7 (https://github.com/jOOQ/jOOQ/issues/13469), and 3.15.11 (https://github.com/jOOQ/jOOQ/issues/13470)

**Comment by lukaseder** [comment]
This seems to work "as designed", according to "parsing ambiguity" on https://www.sqlite.org/lang_upsert.html

It doesn't seem to be a particularly difficult syntax to disambiguate, but I don't know the details of SQLite's parser implementation.
